pkg/apis/k3k.io/v1alpha1/set_types.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+type ClusterSet struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	metav1.TypeMeta   `json:",inline"`
+
+	// Spec is the spec of the ClusterSet
+	Spec ClusterSetSpec `json:"spec"`
+	// Status is the status of the ClusterSet
+	Status ClusterSetStatus `json:"status,omitempty"`
+}
+
+type ClusterSetSpec struct {
```

Comment:
I think we need some way of controlling if the cluster set creates a network policy (especially since the current policy blocks all ingress traffic, essentially making any services inaccessible outside of the cluster). For now maybe we can do a simple enable/disable bool.